Add Nintendo Parental docs (#39118)
home-assistant/home-assistant.github.io: source/_integrations/nintendo_parental.markdown

---
title: Nintendo Switch Parental Controls
description: The Nintendo Switch Parental Controls integration allows you to monitor and control screentime for children via the Nintendo Switch Parental Controls service.
ha_release: 2025.7
ha_iot_class: cloud_polling
ha_codeowners:
  - '@pantherale0'
ha_domain: nintendo_parental
ha_integration_type: integration
ha_platforms:
  - sensor
---

The **Nintendo Switch Parental Controls** {% term integration %} integrates with the Nintendo Switch Parental Controls service, allowing parents to monitor and control screentime for their children.

## Supported devices

Devices supported by the Nintendo Switch Parental Controls mobile app are supported by this integration.

> ## Setup

Reviewer: Please split this section up into **Prerequisites** and **Configuration**.
you can find an example documentation structure on the [examples page](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L43), and in the [developer documentation](https://developers.home-assistant.io/docs/documenting/integration-docs-examples)

the {% include integrations/config_flow.md %} include element renders as follows:

<img width="695" height="318" alt="image" src="https://github.com/user-attachments/assets/1892a404-21d4-42cd-b1f6-40764cb9c143" />

So there is no need to add a Configuration title. It comes with the include element.

Author: I've quickly split them up, but I'm not 100% sure how I should document these steps

 3. You will be prompted to provide an "Access Token". Click the link provided in the dialog description. This link is unique to your setup session.

 4. A new browser tab will open. Log in to your Nintendo account using your credentials.

 5. After successful login, you will see a "Linking an External Account" screen. For the Nintendo account you wish to link, **right-click** on the red button labeled **"Select this person"** and choose **"Copy Link"** (or "Copy Link Address" / "Copy URL" depending on your browser).

    - **Important:** Do **not** click the button directly, as this can redirect you and prevent you from getting the necessary token.

    - The copied link should have a format similar to `npf54789befxxxxxxxx://auth#session_token_code={redacted}&state={redacted}&session_state={redacted}`.

 6. Close the Nintendo Account browser tab.

 7. Paste the entire copied link (the full string) into the "Access Token" field in the Home Assistant configuration dialog.

They seem like quite a lot to add into a field description, but its not so much a prerequisite as the link that is produced in the config flow is uniquely generated each time you start a config flow - that link then also cannot be used in a new config flow instance.

Reviewer: Are these steps on Nintendo side? Then I don't think they belong into this documentation. Can you link to Nintendo documentation instead?

Author: These steps are to setup the integration with Home Assistant. It needs to be in these docs otherwise people won't know how to set the integration up.

There are no docs from Nintendo for this.

Reviewer: I see. But this access token, I need to generate it on the Nintendo site first? So that I can then add it during set up in HA?

Author: You would need to click on the link that will show up in the config flow to generate the token, the module generates that on each new instance and therefore is unique to a specific session.

Reviewer: Ok, I think I'm starting to understand why you didn't split this into prereqs. In that case I think it's fine not to use the standard config_flow text element and document your specific steps.

Author: I suppose the prereqs are probably more "you first need to configure the Nintendo switch with Nintendo's parental control service", then the config flow takes care of most of the authentication process (apart from the bit where the user must click the unique link, login and then copy the token from one of the buttons on the screen).

You just can't get that link until the config flow has already started.

If it were Microsoft (as an example) things are quite different, and a simple button or link will be fine as that is static.

description: The Nintendo Switch Parental Controls integration allows you to monitor and control screentime for children via the Nintendo Switch Parental Controls service.
ha_release: 2025.11
ha_iot_class: cloud_polling
ha_codeowners:
  - '@pantherale0'
ha_domain: nintendo_parental
ha_integration_type: integration
ha_platforms:
  - sensor
  - time
---

The **Nintendo Switch Parental Controls** {% term integration %} integrates with the Nintendo Switch Parental Controls service, allowing parents to monitor and control screentime for their children.

## Supported devices

Devices supported by the Nintendo Switch Parental Controls mobile app are supported by this integration.

## Prerequisites

To use the Nintendo Switch Parental Controls integration, you must access your Nintendo account's authentication tokens. This process must be performed on a device **without** the official Nintendo Switch Parental Controls mobile app installed. The integration needs to intercept the authentication flow, which the app typically handles automatically.

You will need:
- Your Nintendo account credentials.
- A computer or mobile device without the official app installed.

## Configuration

>  1. In the Home Assistant UI, go to **Settings > Devices & Services**.

Reviewer: ```suggestion
 1. In the Home Assistant UI, go to {% my integrations title="**Settings** > **Devices & services**" %}.
```

>  2. Click on **ADD INTEGRATION** and search for "Nintendo Switch Parental Controls".

Reviewer: ```suggestion
 2. Select **Add integration** and search for **Nintendo Switch Parental Controls**.
```
- We prefer [select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click).
- We use bold to markup UI strings.
- Don't use all capitals, even if the UI styling does that. use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization)

>  3. You will be prompted to provide an "Access Token". Click the link provided in the dialog description. This link is unique to your setup session.

Reviewer: ```suggestion
 3. You will be prompted to provide an **Access token**. Select the link provided in the dialog description. This link is unique to your setup session.
```

 4. A new browser tab will open. Log in to your Nintendo account using your credentials.

>  5. After successful login, you will see a "Linking an External Account" screen. For the Nintendo account you wish to link, **right-click** on the red button labeled **"Select this person"** and choose **"Copy Link"** (or "Copy Link Address" / "Copy URL" depending on your browser).

Reviewer: ```suggestion
 5. After successful login, you will see a **Linking an External Account** screen. For the Nintendo account you wish to link, right-click the red button **Select this person** and choose **Copy Link** (or **Copy Link Address** / **Copy URL** depending on your browser).
```

>     - **Important:** Do **not** click the button directly, as this can redirect you and prevent you from getting the necessary token.

Reviewer: ```suggestion
    - **Important**: Do not select the button directly, as this can redirect you and prevent you from getting the necessary token.
```

    - The copied link should have a format similar to `npf54789befxxxxxxxx://auth#session_token_code={redacted}&state={redacted}&session_state={redacted}`.

 6. Close the Nintendo Account browser tab.

>  7. Paste the entire copied link (the full string) into the "Access Token" field in the Home Assistant configuration dialog.

Reviewer: ```suggestion
 7. Paste the entire copied link (the full string) into the **Access Token** field in the Home Assistant configuration dialog.
```

>  8. Click **Submit**.

Reviewer: ```suggestion
 8. Select **Submit**.
```

 9. The configuration flow should then present additional options.

> 10. Click **Submit** to finalize the setup.

Reviewer: ```suggestion
10. Select **Submit** to finalize the setup.
```
